korlib/texture.cpp

```diff
@@ -161,6 +182,90 @@ struct _LevelData
     { }
 };
 
+static inline int _get_num_levels(pyGLTexture* self) {
+    PyObjectRef size = PyObject_GetAttrString(self->m_blenderImage, "size");
+    float width = (float)PyFloat_AsDouble(PySequence_GetItem(size, 0));
+    float height = (float)PyFloat_AsDouble(PySequence_GetItem(size, 1));
+
+    int num_levels = (int)std::floor(std::log2(std::max(width, height))) + 1;
+
+    // Major Workaround Ahoy
+    // There is a bug in Cyan's level size algorithm that causes it to not allocate enough memory
+    // for the color block in certain mipmaps. I personally have encountered an access violation on
+    // 1x1 DXT5 mip levels -- the code only allocates an alpha block and not a color block. Paradox
+    // reports that if any dimension is smaller than 4px in a mip level, OpenGL doesn't like Cyan generated
+    // data. So, we're going to lop off the last two mip levels, which should be 1px and 2px as the smallest.
```

Comment:
So what happens is when Cyan compresses textures, it only creates mip levels down to 4px. But it pads out the remaining layers with garbage data:
https://github.com/H-uru/Plasma/blob/master/Sources/Plasma/PubUtilLib/plGImage/hsDXTDirectXCodec.cpp#L218-L239

When the pipeline loads images, it discards any mip levels smaller than 4px:
https://github.com/H-uru/Plasma/blob/master/Sources/Plasma/PubUtilLib/plPipeline/DX/plDXPipeline.cpp#L8632-L8637